pkg/project/registry/project/proxy/proxy.go

```diff
@@ -156,6 +163,14 @@ func (s *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectI
 
 	s.updateStrategy.PrepareForUpdate(ctx, obj, oldObj)
 	if errs := s.updateStrategy.ValidateUpdate(ctx, obj, oldObj); len(errs) > 0 {
+		// If the user has the ability to update the namespace directly, they can circumvent most of the
```

Comment:
Yes.  As far as the authorizer is concerned, they are completely separate resources.  Almost no one has rights to do anything with namespaces.  Also, almost no one can `create` projects.  This is how the `projectrequest` flow works (you say I want a project, the system makes a project under specific constraints, and then the underlying namespace is created).